agent/agent.go

```diff
@@ -482,6 +482,14 @@ func (a *Agent) setupTask(agentCtx, setupCtx context.Context, initialTC *taskCon
 	tc.taskConfig.WorkDir = tc.taskDirectory
 	tc.taskConfig.Expansions.Put("workdir", tc.taskConfig.WorkDir)
 
+	factory, ok = command.GetCommandFactory("setup.initial")
```

Comment:
I tested it on staging. I had to add this back in because the otherwise it didn't log like it does now. It's because the first one is before startLogging. I can probably remove the first one but I'm too scared it will have some hidden unintended consequence. 

The rest of the logs are all there. The order differs each time tasks are run so it's hard to compare the order to main.

Reply:
the logs on main: https://parsley.mongodb.com/evergreen/evergreen_build_simple_test_patch_1072528ebb0ab31bda6733e16a6ff8bbd19651e7_64df9cada4cf4733f348ed92_23_08_18_16_30_45/1/all?bookmarks=0,1,2,3,4,5,6,7,417

the logs with the changes: 
https://parsley-staging.corp.mongodb.com/evergreen/evg_build_simple_test_patch_1072528ebb0ab31bda6733e16a6ff8bbd19651e7_64dfaa4f9dbe323bd245bf8c_23_08_18_17_28_59/0/all?bookmarks=0,1,2,3,4,5,6,7,213

Reply:
while the order is different between main and with the changes, they are also not in the same order with every run. 
for example, here is another run with the changes, and the order is different than the other staging link above: 

https://parsley-staging.corp.mongodb.com/evergreen/evg_build_simple_test_patch_1072528ebb0ab31bda6733e16a6ff8bbd19651e7_64dfafc19dbe323bd245bf96_23_08_18_17_52_12/0/all?bookmarks=0,210

Reply:
SGTM. Don't really know why the logs would be out of order, but I have noticed before that logs that come in at the same time with the same exact timestamp oftentimes are mixed up.